funsor/terms.py

```diff
@@ -756,51 +756,40 @@ def _(arg, indent, out):
 interpreter.children.register(Funsor)(interpreter.children_funsor)
 
 
-@dispatch(object)
-def to_funsor(x):
+@singledispatch
+def to_funsor(x, output=None, dim_to_name=None):
     """
     Convert to a :class:`Funsor` .
     Only :class:`Funsor` s and scalars are accepted.
 
     :param x: An object.
     :param funsor.domains.Domain output: An optional output hint.
+    :param OrderedDict dim_to_name: An optional inputs hint.
```

Comment:
Could you add a description of the keys and values? In particular:
1. are dims negative or nonnegative or either?
2. are names `str` (as I would expect) or `Tuple[str, Domain]` as commented above?